### ISSUE TITLE
AttributeError: 'UnicodeReader' object has no attribute 'next' in dwca.py

### ISSUE THREAD

**Issue by danstoner** [iDigBio/idb-backend#151]
While running `db-check`.

Not caught by test suite.

```
UUID=c38b867b-05f3-4733-802e-d8d2d3324f84 TYP=db-check bash -c 'idigbio-ingestion -v --logfile ./$TYP_$UUID.`date +"%Y-%m-%d_%H%M%S"`.log $TYP $UUID'
2021-04-05 16:29:44.994 INFO  idb.db-check.c38b867b-05f3-4733-802e-d8d2d3324f84჻ Starting db_check ingest: False
2021-04-05 16:29:45.532 INFO  idb.db-check.c38b867b-05f3-4733-802e-d8d2d3324f84჻ Processing c38b867b-05f3-4733-802e-d8d2d3324f84, type: application/zip
2021-04-05 16:29:47.713 ERROR idb.db-check჻ Unhandled Exception when processing c38b867b-05f3-4733-802e-d8d2d3324f84
Traceback (most recent call last):
  File "/home/dan/git/IDIGBIO/idb-backend/idigbio_ingestion/db_check.py", line 443, in process_file
    dwcaobj = Dwca(fname, skipeml=True, logname="idb")
  File "/home/dan/git/IDIGBIO/idb-backend/idigbio_ingestion/lib/dwca.py", line 92, in __init__
    logname=self.logger.name)
  File "/home/dan/git/IDIGBIO/idb-backend/idigbio_ingestion/lib/dwca.py", line 208, in __init__
    self._reader.next()
AttributeError: 'UnicodeReader' object has no attribute 'next'
2021-04-05 16:29:47.717 INFO  idb.db-check჻ c38b867b-05f3-4733-802e-d8d2d3324f84 writing summary json
2021-04-05 16:29:47.876 INFO  idb.db-check.c38b867b-05f3-4733-802e-d8d2d3324f84჻ Finished db_check in 2.879s

```

**Comment by danstoner** [comment]
The python3 version of `unicodecsv` does not seem to have a `next` function.

```
$ grep next venv/lib/python2.7/site-packages/unicodecsv/py3.py
    def __next__(self):
        return self.reader.__next__()

$ grep next venv/lib/python2.7/site-packages/unicodecsv/py2.py
    >>> row = r.next()
    def next(self):
        row = self.reader.next()
    >>> r.next() == {'a': u'1', u'ñ':'2', 'r': [u'î']}
    >>> r.next() == {'a': u'1', u'ñ':'2', 'r': [u'\xc3\xb8']}
    >>> r.next() == {'a': u'\xc3\xa9', u'ñ':'2', 'r': [u'\xc3\xae']}
    >>> print r.next() == {'name': 'Cary Grant', 'place': 'hollywood'}
    >>> print r.next() == {'name': 'Nathan Brillstone', 'place': u'øLand'}
    >>> print r.next() == {'name': u'Will ø. Unicoder', 'place': u'éSpandland'}
            self.fieldnames = _stringify_list(reader.next(), reader.encoding)
    def next(self):
        row = csv.DictReader.next(self)
```

The python3 lib wants to implement the iterator protocol.

**Comment by danstoner** [comment]
The `next` issue resolved via a90f2d2406efd8f6aa028e9a3099617e12c21ee6, but now I will continue to iterate on issues in db-check...

```
2021-04-06 11:49:30.401 ERROR idb.db-check჻ Unhandled Exception when processing c38b867b-05f3-4733-802e-d8d2d3324f84
Traceback (most recent call last):
  File "/home/dan/git/IDIGBIO/idb-backend/idigbio_ingestion/db_check.py", line 447, in process_file
    dwcrf, rsid, existing_etags, existing_ids, ingest=ingest, db=db)
  File "/home/dan/git/IDIGBIO/idb-backend/idigbio_ingestion/db_check.py", line 395, in process_subfile
    eu_set = existing_etags.viewkeys()
AttributeError: 'dict' object has no attribute 'viewkeys'
```

**Comment by danstoner** [comment]
db-check looks like it is working since 641d4bcefbc028788afa73cd78f9ad5d75f0cf5e.
